### PR TITLE
task 6 (complete beginner in js) completed; debugger: Debugger for Chrome

### PR DESCRIPTION
**Pull request by mshevko**

### CODE REVIEW

File: task6/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (MaksimViarenich) [inline task6/.vscode/launch.json:4]: JSON doesn't allow comments

File: task6/script.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+        postsList.push(
+            {
+                id: (i + 1).toString(),
+                description: 'post ' + (i + 1) + ' created. successfully or not? who knows.',
+                createdAt: new Date(),
+                author: 'author ' + (i + 1),
+                hashTags: ['tag ' + (i + 1), 'tag ' + (i + 2), 'tag ' + (i + 3)],
+                likes: [{author: 'author ' + (i + 10)}, {author: 'author ' + (i + 12)}, {author: 'author ' + (i + 14)}],
+            }
+        )
```
Comment (MaksimViarenich) [inline task6/script.js:15]: Don't forget about ; after command

File: task6/script.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                        case 'dateBefore':
+                            arrCopy = arrCopy.filter(post => post.createdAt <= filterConfig.dateBefore);
+                            break;
+                        case 'hashTags':
+                            let res = [];
+
+                            filterConfig.hashTags.forEach(
+                                elem =>
+                                {res = res.concat(arrCopy.filter(post => post.hashTags.includes(elem)))}
+                            )
```
Comment (MaksimViarenich) [inline task6/script.js:73]: Add ;

File: task6/script.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        } catch (msg) {
+            console.error(msg);
+            return;
+        }
+
+        var keys = Object.keys(post);
+        
+        if (keys.includes('id') === false || typeof post['id'] !== 'string') {
+            return false;
+        }
```
Comment (MaksimViarenich) [inline task6/script.js:135]: Minor. After `if` we add empty line

File: task6/script.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        } catch (msg) {
+            console.error(msg);
+            return false;
+        }
+
+        if (validatePost(post) && (postsList.findIndex(elem => elem.id === post.id) === -1)) {
+            postsList.push(post);
+            return true;
+        }
+        else {
```
Comment (MaksimViarenich) [inline task6/script.js:175]: `else` is not needed. You can just write `return false;`

File: task6/.vscode/launch.json (hunk 1 above)
Comment (mshevko) [reply task6/.vscode/launch.json:4]: sorry, missed this, cuz the comments where generated by idea. i'll delete them.

File: task6/script.js (hunk 5 above)
Comment (mshevko) [reply task6/script.js:175]: am i right, that as soon as i'll fix all the code style problems i can merge this branch without asking for your second approval?

Comment (MaksimViarenich) [reply task6/script.js:175]: If you want to fix them and increase your mark, you have to fix it and ask for second approval.

Comment (mshevko) [reply task6/script.js:175]: o'key, i'll do this right now

File: task6/script.js (hunk 4 above)
Comment (mshevko) [reply task6/script.js:135]: after all the `if`-s in the code or only in this case?

Comment (MaksimViarenich) [reply task6/script.js:135]: If after `if` you have some code, you should add an empty line

**Comment by MaksimViarenich** [comment]
You can merge it

**Pull request merged**